projects/ngx-datatable/src/lib/components/body/body-row.component.ts

```diff
@@ -232,4 +232,14 @@ export class DataTableBodyRowComponent<TRow = any> implements DoCheck, OnChanges
   onTreeAction() {
     this.treeAction.emit();
   }
+
+  /** Returns the row index, or if in a group, the index within a group. */
+  private get innerRowIndex(): number {
+    if (typeof this.rowIndex === 'number') {
+      return this.rowIndex;
+    }
+
+    const [, innerIndex] = this.rowIndex.match(/\d+-(\d+)/);
```

Comment:
Would it make sense to not use a string as a group - group item index but use two entries in a structure like groupIndex and rowIndex. I do not like the regex approach here.

Reply:
not using a string will be a breaking change as we emit this type. While I would love to do this, I think we should wait and first just have the types.